src/silx/math/colormap.py

```diff
@@ -321,6 +321,15 @@ def autoscale_mean3std(self, data):
             mean + 3 * std, 0.0, 1.0
         )
 
+    def autoscale_percentile_1_99(self, data):
+        """Autoscale using [1st, 99th] percentiles"""
+        data = data[self.is_valid(data)]
+        if data.dtype.kind == "f":  # Strip +/-inf
+            data = data[numpy.isfinite(data)]
+        if data.size == 0:
+            return None, None
+        return numpy.nanpercentile(data, (1, 99))
```

Comment:
Use a method to compute percentile as for the other autoscale modes.
This allows to compute the percentile only on data that is valid for the current normalization mode (e.g. logarithmic) and to ignore +/-inf values.